apps/cowswap-frontend/src/common/hooks/useModalState.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react'
+
+export interface ModalState<T> {
+  isModalOpen: boolean
+  openModal: (context?: T) => void
+  closeModal: () => void
+  context?: T
+}
+
+export function useModalState<T>(trigger?: boolean): ModalState<T> {
+  const [context, setContext] = useState<T | undefined>(undefined)
+  const [isModalOpen, setIsModalOpen] = useState(false)
+
+  const openModal = (context?: T) => {
+    setIsModalOpen(true)
+    setContext(context)
+  }
+  const closeModal = () => {
+    setIsModalOpen(false)
+    setContext(undefined)
+  }
+
+  useEffect(() => {
+    if (trigger === undefined) return
+
+    if (trigger) {
+      openModal()
+    } else {
+      closeModal()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why, is it re-rendering because of the open/close modal handlers? 
Can't you use useCallback for them?

Reply:
Thank you!
Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/1de82e6a749f6fbf39d93b92931dd28690baf08e